lib/src/default_revset_engine.rs

```diff
@@ -565,6 +566,12 @@ struct EvaluationContext<'index> {
     composite_index: CompositeIndex<'index>,
 }
 
+fn to_u32_generation_range(range: &Range<u64>) -> Range<u32> {
+    let start = range.start.try_into().unwrap();
+    let end = range.end.try_into().unwrap_or(u32::MAX);
```

Comment:
nit: I would make both start/end do `.unwrap_or(u32::MAX)` assuming there would be no linear history containing `u32::MAX` entries.

If we add a revset function/argument to specify generation, it's possible to specify unreasonably large `start` value.

Reply:
I was thinking that the `unwrap_or()` on the upper bound was basically to handle the default "infinite" range, but that's a good point about future support for letting the user specify the generation range. I made this case error out gracefully instead.

Reply:
Yes, it's mainly for the infinite range, but also works for future extension. For the lower bound, erroring out should also be good. Thanks.